patches/minecraft/net/minecraft/entity/item/EntityItem.java.patch

```diff
@@ -68,23 +76,36 @@
                      else
                      {
                          itemstack1.func_190917_f(itemstack.func_190916_E());
-@@ -282,6 +303,7 @@
+@@ -282,16 +302,16 @@
  
      public boolean func_70097_a(DamageSource p_70097_1_, float p_70097_2_)
      {
 +        if (this.field_70170_p.field_72995_K || this.field_70128_L) return false; //Forge: Fixes MC-53850
          if (this.func_180431_b(p_70097_1_))
          {
              return false;
-@@ -314,6 +336,7 @@
+         }
+-        else if (!this.func_92059_d().func_190926_b() && this.func_92059_d().func_77973_b() == Items.field_151156_bN && p_70097_1_.func_94541_c())
```

Comment:
It might be better to keep this logic here and pass it to the Forge hook as a boolean parameter, rather than reimplementing it in the hook.

As this is only a small amout of logic, it's not essential, but it tends to make patches like this more likely to be ported correctly in the event of vanilla changes.

Reply:
Could I move the default cancellation action to the attackEntityFrom method instead maybe?

Having a boolean for a single very, very, very rare occasion seems pretty wasteful..